Add new InlineSupportLink format to Calypso sections

jjchrisdiehl opened the pull request:
#### Changes proposed in this Pull Request

This PR implements the new `supportContext` format for the InlineSupportLinks component across various Calypso screens. 

#### Deprecated usage:
```js
import InlineSupportLink from 'calypso/components/inline-support-link';
function Link() {
	const inlineSupportProps = {
		supportLink: 'https://wordpress.com/support/audio/podcasting/',
		supportPostId: 38147,
}
```

#### New Example Usage

```js
import InlineSupportLink from 'calypso/components/inline-support-link';
function Link() {
	return <InlineSupportLink supportContext="purchases" showIcon={ false } />
}
```

The `supportContext` is a combination of the `supportPostId` and `supportLink` now found in _context-links.js_.

```js
const getContextLinks = () => ( {
	purchases: {
		link: 'https://wordpress.com/support/manage-purchases/',
		post_id: 111349,
	}
} );
```

#### Testing instructions

This PR relies on the merger of #54991, once merged please test the following:
<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

Test that the existing deprecated format still works:

- Open https://wordpress.com/settings/writing/
- Go to Podcasting
- Click on Learn More
- Modal should load with Podcasting support documentation

Test that the new format works too:
- Go to Posts or Pages
- Click on Learn More at the top of the section
- Modal should load with respective support documentation

<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

#### For translators

Strings that look like:

`Create, edit, and manage the posts on your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`

Will render as a subheader:
![image](https://user-images.githubusercontent.com/16580129/128423872-89ae0721-38ac-4df0-a613-742b23a2eec4.png)

Strings that look like (omit the Learn More text):

`Publicize posts {{learnMoreLink/}}`

Will render as an icon:
![image](https://user-images.githubusercontent.com/16580129/128424013-b9455f61-a836-4e35-8fe5-3fa2417b0a3a.png)


Related to #54991

Commit:
Add Learn More link to Posts page

Commit:
Add Learn More link to Pages section

Commit:
Add Learn More link to Email section

Commit:
Add Learn More link to Stats section

Commit:
Add Learn More link to Domains section

Commit:
Add Learn More link to Media section

Commit:
Add Learn More link to Portfolio and Testimonial sections

Commit:
Add new format for InlineSupportLinks to page section

Commit:
Update Posts section to new support link format

Commit:
Update Stats section to new support link format

Commit:
Update Domains section to new support link format

Commit:
Update Emails section to new support link format

Commit:
Update Media section to new support link format

Commit:
Update portfolio/testimonial to new support format

Commit:
Update Comments section to new support link format

Commit:
Update Themes section to new support link format

Commit:
Update Publicize section to new support link format

Commit:
Update Ads section to new support link format

Commit:
Update Import section to new support link format

Commit:
Update Export section to new support link format

Commit:
Update Team section to new support link format

Commit:
Update Privacy section to new support link format

Commit:
Update Discussion section to new support link format

Commit:
Update Plugins section to new support link format

Commit:
Update Plugin/Manage section to new support format

Commit:
Update Purchases section to new support link format

a8ci18n commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -575,7 +576,12 @@ export class SiteSettingsFormGeneral extends Component {
 					onButtonClick={ handleSubmitForm }
 					showButton
-					title={ translate( 'Privacy', { context: 'Privacy Settings header' } ) }
+					title={ translate( 'Privacy {{learnMoreLink}}{{/learnMoreLink}}', {
```
:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

Aurorum replied:
Same as above.

Aurorum commented on client/my-sites/marketing/connections/connections.jsx:
```diff
@@ -19,7 +20,14 @@ const SharingConnections = ( { translate } ) => (
-		<SharingServicesGroup type="publicize" title={ translate( 'Publicize posts' ) } />
+		<SharingServicesGroup
+			type="publicize"
+			title={ translate( 'Publicize posts {{learnMoreLink}}{{/learnMoreLink}}', {
```
Is this meant to have a "Learn more" link too? If not, I think it could be re-written like this which could be more consistent to similar scenarios in Calypso:

```suggestion
			title={ translate( 'Publicize posts {{learnMoreLink/}}', {
```

Aurorum commented on client/my-sites/pages/main.jsx:
```diff
@@ -76,7 +77,7 @@ class PagesMain extends React.Component {
 		const { siteId, search, status, translate, queryType, author } = this.props;
 		const postStatus = mapPostStatus( status );
-
+		const allSites = siteId ? 1 : 2;
```
I'm guessing you didn't mean to commit this line!

sirbrillig replied:
🤔 I imagine this was intended to be the condition to be used to decide if the `subHeaderText` below should be singular or plural, although the variable name doesn't necessarily make that clear. It does, in fact, mirror the existing translation which uses `siteId` for that same condition.

Maybe refactoring this slightly would make it easier to read by removing this line and down below specifying something like:

```
//...
count: siteId ? 1 : 0
//...
```

In this way you're using `0` when the page has no single site selected, which should (I hope) still provide the plural version of the translation without having to resort to the otherwise arbitrary `2`.

michaeldcain replied:
I think we settled with `allSites` for the naming b/c the plural subheader is for the "All Sites Mode". Maybe just a comment above this line explaining the variable would be enough.

Aurorum replied:
Ah, apologies - looking at this again, I see what's going on here now. I assumed it was for testing the pluralised translations, but I didn't notice the check for `siteId` was removed.

Aurorum commented on client/my-sites/posts/main.jsx:
```diff
@@ -55,6 +56,7 @@ class PostsMain extends React.Component {
 	render() {
 		const { author, category, search, siteId, statusSlug, tag, translate } = this.props;
 		const status = mapPostStatus( statusSlug );
+		const allSites = siteId ? 1 : 2;
```
Same as above - I'm guessing you didn't mean to commit this line and it was just for testing.

Commit:
Add corrections from PR convo #55078

jjchrisdiehl commented:
@michaeldcain The above comments should be resolved [in this commit](https://github.com/Automattic/wp-calypso/pull/55078/commits/8007d1f67d9aa09bc10930bfad58918adbb48250), could you please take a look.

I added a comment above the `allSites` variable and modified it to `isAllSites` to provide a bit more clarity. I also incorporated @sirbrillig suggestion to use `siteID ? 1 : 0`.

Additionally, fixed the component syntax as pointed out by @Aurorum

a8ci18n commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -575,7 +576,12 @@ export class SiteSettingsFormGeneral extends Component {
 					onButtonClick={ handleSubmitForm }
 					showButton
-					title={ translate( 'Privacy', { context: 'Privacy Settings header' } ) }
+					title={ translate( 'Privacy {{learnMoreLink/}}', {
```
:information_source: This string doesn't exist elsewhere without a **context** (or with a different one). You can probably drop the context. If needed, you can use a translator comment instead to clarify meaning.

michaeldcain replied:
@jjchrisdiehl: I know you didn't add it, but it looks like you can switch the `context` to a translator `comment`.

jjchrisdiehl replied:
Updated this here https://github.com/Automattic/wp-calypso/pull/55078/commits/929eab25ba39f85b11ba715ca8a6ba7d8de1afee

Waiting on tests to complete

michaeldcain commented on client/my-sites/domains/domain-management/list/index.jsx:
```diff
@@ -131,7 +132,13 @@ export class List extends React.Component {
 						className="domain-management__page-heading"
 						headerText={ translate( 'Site Domains' ) }
-						subHeaderText={ translate( 'Manage the domains connected to your site.' ) }
+						subHeaderText={ translate( 'Manage the domains connected to your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.',
```
It looks like the code linter wants `'Manage the domains connected to your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.',` on its own line.

Commit:
Update translator context to comment

nbloomf reviewed (approved):
Looks good; I clicked through all the links to make sure they work and point to the right place.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6331131

Hi @jjchrisdiehl, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `View, manage, or cancel your plan and other purchases. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `View, reply to, and manage all the comments across your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Manage the domains connected to your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Explore tools to earn money with your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Your home base for accessing, setting up, and managing your emails. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Back up or move your content to another site or platform. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Import content from another website or platform. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Publicize posts {{learnMoreLink/}}`
- `Manage all the media on your site, including images, video, and more. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Create, edit, and manage the pages on your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Invite contributors to your site and manage their access settings. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Add new functionality and integrations to your site with plugins. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Create, edit, and manage the posts on your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Privacy {{learnMoreLink/}}`
- `Control how people interact with your site through comments. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Learn more about the activity and behavior of your site's visitors. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Select or update the visual design for your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Create and manage all the testimonials on your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`
- `Create, edit, and manage the portfolio projects on your site. {{learnMoreLink}}Learn more{{/learnMoreLink}}.`

Thank you in advance!

simison commented:
Can you elaborate a bit more on why of this PR, it's very elaborate on what already but I wanted to better understand this. :-) 

cc @razvanpapadopol @autumnfjeld might be relevant for your work.

Would it be useful to make this `@automattic/components` component and use it also in wp-admin/editor?

simison commented:
Fabulous, thanks for explaining!

Commit:
Update PropTypes for title of SharingServicesGroup

Commit:
Add import Experiment, accidentally removed

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6331131).
